qiskit_dynamics/pulse/dynamics_job.py

```diff
@@ -0,0 +1,80 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""This module implements the job class used for PulseSimulator objects."""
+
+from typing import Optional, Callable, Dict
+import numpy as np
+
+from qiskit.providers.backend import Backend
+from qiskit.providers import JobV1 as Job
+from qiskit.providers import JobStatus, JobError
+
+
+class DynamicsJob(Job):
+    """DynamicsJob class for PulseSimulator."""
+
+    _async = False
+
+    def __init__(self,
+        backend: Backend,
+        job_id: str,
+        fn: Callable,
+        fn_kwargs: Dict,
+        **kwargs
+    ) -> None:
+        """Initializes the job.
+
+        Args:
+            backend: The backend used to run the job.
+            job_id: A unique id in the context of the backend used to run the job.
+            fn: Function to run the simulation.
```

Comment:
Should we also have some form of callback to track the progress of the simulation? That would be nice.

Reply:
This is related to above: for this PR no but could be a future improvement.